tests/vmi_cloudinit_test.go

```diff
@@ -548,38 +547,33 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
 			})
 
 			DescribeTable("[test_id:3187]should have cloud-init userdata and network-config from separate k8s secrets", func(userDataLabel string, networkDataLabel string) {
-				vmi := tests.NewRandomVMIWithEphemeralDiskAndConfigDriveUserdataNetworkData(
-					cd.ContainerDiskFor(cd.ContainerDiskCirros), "", "", false)
+				uSecretID := fmt.Sprintf("%s-test-secret", uuid.NewString())
+				nSecretID := fmt.Sprintf("%s-test-secret", uuid.NewString())
+
+				vmi := libvmifact.NewCirrosNoUserData(
+					libvmi.WithCloudInitConfigDriveUserDataSecretName(uSecretID),
+					libvmi.WithCloudInitConfigDriveNetworkDataSecretName(nSecretID),
+				)
 
 				idx := 0
-				for i, volume := range vmi.Spec.Volumes {
-					if volume.CloudInitConfigDrive == nil {
-						continue
+				for idx = range vmi.Spec.Volumes {
+					if vmi.Spec.Volumes[idx].CloudInitConfigDrive != nil {
+						break
 					}
-					idx = i
-
-					uSecretID := fmt.Sprintf("%s-test-secret", uuid.NewString())
-					spec := volume.CloudInitConfigDrive
-					spec.UserDataSecretRef = &kubev1.LocalObjectReference{Name: uSecretID}
-
-					nSecretID := fmt.Sprintf("%s-test-secret", uuid.NewString())
-					spec.NetworkDataSecretRef = &kubev1.LocalObjectReference{Name: nSecretID}
-
-					// Store cloudinit data as k8s secret
-					By("Creating a secret with userdata")
-					uSecret := libsecret.New(uSecretID, libsecret.DataString{userDataLabel: testUserData})
+				}
 
-					By("Creating a secret with network data")
-					nSecret := libsecret.New(nSecretID, libsecret.DataString{networkDataLabel: testNetworkData})
+				// Store cloudinit data as k8s secret
+				By("Creating a secret with userdata")
+				uSecret := libsecret.New(uSecretID, libsecret.DataString{userDataLabel: testUserData})
 
-					_, err := virtClient.CoreV1().Secrets(vmi.Namespace).Create(context.Background(), uSecret, metav1.CreateOptions{})
-					Expect(err).ToNot(HaveOccurred())
+				By("Creating a secret with network data")
+				nSecret := libsecret.New(nSecretID, libsecret.DataString{networkDataLabel: testNetworkData})
 
-					_, err = virtClient.CoreV1().Secrets(vmi.Namespace).Create(context.Background(), nSecret, metav1.CreateOptions{})
-					Expect(err).ToNot(HaveOccurred())
+				_, err := virtClient.CoreV1().Secrets(vmi.Namespace).Create(context.Background(), uSecret, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
 
-					break
-				}
+				_, err = virtClient.CoreV1().Secrets(vmi.Namespace).Create(context.Background(), nSecret, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
With this change the By statements indicate when the secret object is instantiated instead of actually be created by the client, in case of an issue it would be confusing to debug.

Could you please move the client create operations closer to the secret instantiation? (move line 572 right after 567)